src/main/java/com/puppycrawl/tools/checkstyle/checks/AbstractTypeAwareCheck.java

```diff
@@ -362,11 +364,15 @@ protected final String getCurrentClassName() {
      */
     protected final AbstractClassInfo createClassInfo(final Token name,
                                               final String surroundingClass) {
+        final AbstractClassInfo result;
```

Comment:
in this particular case `final AbstractClassInfo classInfo = findClassAlias(name.getText());` should be moved inside ELSE block.

in general  I prefer to keep non initialized (result) variable on the top.